File: starter-bots/ReferenceBot/ReferenceBot/Models/GameObject.cs
using System;
using System.Collections.Generic;
using NETCoreBot.Enums;

namespace NETCoreBot.Models
{
    public class GameObject
    {
        public Guid Id { get; set; }
        public int Size { get; set; }
        public int Speed { get; set; }
        public ObjectTypes GameObjectType { get; set; }
        public int CurrentHeading { get; set; }
        public Position Position { get; set; }
        public int TorpedoSalvoCount { get; set; }

        public static GameObject FromStateList(Guid id, List<int> stateList)
        {
            if (stateList.Count == 8)
            {
                return new GameObject
                {
                    Id = id,
                    Size = stateList[0],
                    Speed = stateList[1],
                    CurrentHeading = stateList[2],
                    GameObjectType = (ObjectTypes) stateList[3],
                    Position = new Position {X = stateList[4], Y = stateList[5]},
                    TorpedoSalvoCount = stateList[7]
                };
            }
            else
            {
                return new GameObject
                {
                    Id = id,
                    Size = stateList[0],
                    Speed = stateList[1],
                    CurrentHeading = stateList[2],
                    GameObjectType = (ObjectTypes) stateList[3],
                    Position = new Position {X = stateList[4], Y = stateList[5]}
                };
            }
        }
    }
}

File: starter-bots/KotlinBot/src/main/kotlin/galaxio/model/GameObject.kt
package galaxio.model

import galaxio.model.enums.ObjectTypes
import java.util.UUID

data class GameObject(
	val id: UUID,
	val size: Int,
	val speed: Int,
	val currentHeading: Int,
	val position: Position,
	val gameObjectType: ObjectTypes
) {
	companion object {
		fun fromStateList(entry: Map.Entry<String, List<Int>>): GameObject =
			fromStateList(UUID.fromString(entry.key), entry.value)

		fun fromStateList(id: UUID, stateList: List<Int>): GameObject {
			val position = Position(stateList[4], stateList[5])
			return GameObject(id, stateList[0], stateList[1], stateList[2], position, ObjectTypes.valueOf(stateList[3]))
		}
	}
}